source/code/source/AsyncOp.h

```diff
@@ -0,0 +1,371 @@
+#pragma once
+

```

Comment:
Note that these classes are extremely similar to the ones I wrote for XSAPI last year.  The implementations have changed slightly, but the AsyncOp<> interface remains the same.